### DISCUSSION TITLE
sorting settings only for particular directory

### DISCUSSION THREAD

**Discussion by bobblkabb** [sayanarijit/xplr#609] [Q&A]
Got settings:
```
xplr.config.general.initial_sorting = {
    { sorter = "ByIsDir", reverse = true },
    { sorter = "ByLastModified", reverse = true },
}
```
but I'd like to have them only for a certain directory. Is this possible? How?

version: 0.20.2

**Comment by sayanarijit** [comment]
You need to use the `on_directory_change` hook.

**Reply by bobblkabb** [reply]
Sorry but I don't get it how it works. Documentation didn't help. I was searching the internet and the issues.
Could you please explain it?

**Reply by sayanarijit** [reply]
Sorry for the late reply. Search for on_directory_change in the docs or the default init.lua comments. You'll see how to setup "hooks". Basically you'll have to setup a hook that will: when there's a change in directory, backup the previous sorters, and apply new sorters based on your preferences, then ExplorePwdAsync to refresh the buffer.

You can also have it search for a special file like ".sorters.json/yaml" inside the current directory to add custom sorting from the file path.
